azurerm/internal/services/apimanagement/api_management_user_data_source.go

```diff
@@ -20,7 +20,10 @@ func dataSourceArmApiManagementUser() *schema.Resource {
 		},
 
 		Schema: map[string]*schema.Schema{
-			"user_id": azure.SchemaApiManagementUserDataSourceName(),
+			"user_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
the validation function above uses a regex rather than a Resource ID here:

```
// from the portal: `The field may contain only numbers, letters, and dash (-) sign when preceded and followed by number or a letter.`
	if matched := regexp.MustCompile(`(^[a-zA-Z0-9])([a-zA-Z0-9-]{1,78})([a-zA-Z0-9]$)`).Match([]byte(value)); !matched {
		errors = append(errors, fmt.Errorf("%q may only contain alphanumeric characters and dashes up to 80 characters in length", k))
	}
```

would it be possible to give an example of the value that's disallowed? rather than removing this validation it'd be worth updating it to ensure that it's correct

Reply:
The example is the admin user that is created with user_id  "1" by default. Looking at the API documentation I could not find the restriction on 3-80 characters. Fiddling with the settings I found that the length could be between 1 and 80. 

Will restore the orignal validation and change the regex.